homeassistant/components/automation/numeric_state.py

```diff
@@ -37,13 +38,18 @@ def trigger(hass, config, action):
                       CONF_BELOW, CONF_ABOVE)
         return False
 
+    if value_template is not None:
+        renderer = lambda value: template.render(hass, value_template, value)
+    else:
+        renderer = None
```

Comment:
If you change this to be `lambda value: value.state` you can remove a bunch of logic below. Even can convert `_in_range` to only take value, range_start, range_end.